x-pack/platform/plugins/shared/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/components/steps/components/package_policy_input_stream.tsx

```diff
@@ -324,6 +330,72 @@ export const PackagePolicyInputStreamConfig = memo<Props>(
                   </EuiFlexItem>
                   {isShowingAdvanced ? (
                     <>
+                      {packageInfo.type === 'input' && (
+                        <EuiFlexItem>
+                          <EuiFormRow
+                            label={
+                              <FormattedMessage
+                                id="xpack.fleet.createPackagePolicy.stepConfigure.packagePolicyDataStreamTypeInputLabel"
+                                defaultMessage="Data Stream Type"
+                              />
+                            }
+                            helpText={
+                              isEditPage ? (
+                                <FormattedMessage
+                                  id="xpack.fleet.createPackagePolicy.stepConfigure.packagePolicyInputOnlyEditDataStreamTypeHelpLabel"
+                                  defaultMessage="The data stream type cannot be changed for this integration. Create a new integration policy to use a different input type."
+                                />
+                              ) : (
+                                <FormattedMessage
+                                  id="xpack.fleet.createPackagePolicy.stepConfigure.packagePolicyDataStreamTypeHelpLabel"
+                                  defaultMessage="Select a data stream type for this policy. This setting changes the name of the integration's data stream. {learnMore}."
+                                  values={{
+                                    learnMore: (
+                                      <EuiLink
+                                        href={docLinks.links.fleet.datastreamsNamingScheme}
+                                        target="_blank"
+                                      >
+                                        {i18n.translate(
+                                          'xpack.fleet.createPackagePolicy.stepConfigure.packagePolicyNamespaceHelpLearnMoreLabel',
+                                          { defaultMessage: 'Learn more' }
+                                        )}
+                                      </EuiLink>
+                                    ),
+                                  }}
+                                />
+                              )
+                            }
+                          >
+                            <EuiRadioGroup
+                              data-test-subj="packagePolicyDataStreamType"
+                              disabled={isEditPage}
+                              idSelected={packagePolicyInputStream.data_stream.type || 'logs'}
```

Comment:
The Package Spec requires a type now in policy templates in input packages. 

Should we remove this requirement, now that the type can be configured?

As it is now here, this code will support streams without defined type, and will use `logs` as default.

Reply:
Yes we can probably remove this if we bump the max spec version available in kibana